src/toil/jobStores/aws/jobStore.py

```diff
@@ -413,17 +424,24 @@ def _importFile(self, otherCls, url, sharedFileName=None, hardlink=False):
                 info.save()
             finally:
                 srcKey.bucket.connection.close() 
-            return FileID(info.fileID, size) if sharedFileName is None else None
+            return FileID(info.fileID, size, executable) if sharedFileName is None else None
         else:
+            #print(f"CHECKING AWS: {srcKey.path}")
+            print("IN ELSE")
             return super(AWSJobStore, self)._importFile(otherCls, url,
                                                         sharedFileName=sharedFileName)
 
     def _exportFile(self, otherCls, jobStoreFileID, url):
+        print("IN AWS _EXPORTFILE")
+        print(f"otherCls: {otherCls}")
+        print(f"AWS url: {url}")
         if issubclass(otherCls, AWSJobStore):
             dstKey = self._getKeyForUrl(url)
+            print(f"AWS jobStoreFileID.executable: {jobStoreFileID.executable}")
             try:
                 info = self.FileInfo.loadOrFail(jobStoreFileID)
                 info.copyTo(dstKey)
```

Comment:
@adamnovak @DailyDreaming When I was trying to take the executable attribute from a `FileID` to an AWS key during `_exportFile()`, I was running into issues getting access to it. I saw that the `AWS jobstore` uses the FileInfo class to preserve the `FileID` attributes which gets copied over to the new AWS key, so I was wondering if I should maybe add an executable attribute on to that class so it gets copied over too.

FileInfo class: https://github.com/DataBiosphere/toil/blob/a03e24f7e8afbfbd6416c17c6f906b6fdf6435a7/src/toil/jobStores/aws/jobStore.py#L828 

Sorry for all the extra print statements, I just wanted to keep them there for reference until I finish debugging this part.

Reply:
If you did work out a way to store the executable bit in the AWS FileInfo, then you would have the AWSJobStore be responsible for keeping track of that bit, probably instead of making the FileID do it.

That would let you set the executable bit properly when people give you strings instead if FileIDs when they download files, but it would be a different design than the one you've been implementing so far, where the FileID keeps track of executability, so I wouldn't do it.

These `_importFIle` and `_exportFile` overrides in `AWSJobStore` have one purpose, which is to make copying more efficient when you are importing from something on AWS to an `AWSJobStore`, or exporting from an `AWSJobStore` to somewhere else on AWS. Instead of downloading the file from AWS to the local machine and then uploading it elsewhere on AWS, they detect AWS-to-AWS operations and ask AWS to just copy the data internally.

Since files on AWS have no concept of executability, there's no execute bit to read when you import from AWS, and no execute bit to set when you export to AWS. So you shouldn't need to touch the AWS `_importFile` and `_exportFile` overrides in any substantial way; they should just not set the execute bit in the FileIDs they produce for imports from AWS, and not do anything with the execute bit in the file IDs they get to export to AWS.

If you are having trouble with imports and exports between AWS and local files, you might need to look at `_importFile` and `_exportFile` as defined in the `AbstractJobStore` (which is being called from here when the other side of the copy is not on AWS), or in the `FileJobStore`.

Does that help?